lib/elastic/helpers.js

```diff
@@ -0,0 +1,297 @@
+'use strict';
+
+const db = require('../services/db'),
+  _ = require('lodash'),
+  bluebird = require('bluebird'),
+  refProp = '_ref',
+  log = require('../services/log').withStandardPrefix(__dirname);
+
+/**
+ * Used to test types
+ *
+ * @type {Array}
+ */
+var genericTypeParsers = [{
+  when: compare('string'),
+  then: convertOpValuesPropertyToString
+}, {
+  when: compare('date'),
+  then: convertOpValuesPropertyToDate
+}, {
+  when: compare('object'),
+  then: function (propertyName, ops) {
+    return ops;
+  }
+}];
+
+// Required for `_.listDeepObjects`
+_.mixin(require('lodash-ny-util'));
+
+/**
+ * A function which returns a function to help
+ * compare two values. Pass in the value you exect
+ * and then invoke the function with the value you're
+ * testing as the new argument.
+ *
+ * @param  {String} expected
+ * @return {Function}
+ */
+function compare(expected) {
+  return function (comparison) {
+    return expected === comparison;
+  };
```

Comment:
Arrow function could be nice here `return comparison => expected === comparison;`